src/components/ExperienceForm/ExperienceForm.js

```diff
@@ -3,155 +3,222 @@
+      error={!!errors.body && touched.body}
+      disabled={isSubmitting}
+      className={classes.textField}
+    />
+    <Button
+      variant="raised"
+      color="primary"
+      type="submit"
+      disabled={isSubmitting}
+      className={classes.button}
+    >
+      {FORM_BUTTON_INSERT_UPDATE}
+    </Button>
+  </form>
+);
 
-  static defaultProps = {
-    submitHandler: false,
-    experienceSlug: false,
-    experiences: {
-      error: null,
-      items: []
-    }
-  };
+ExperienceForm.propTypes = {
+  handleSubmit: PropTypes.func.isRequired,
+  handleChange: PropTypes.func.isRequired,
+  handleBlur: PropTypes.func.isRequired,
+  isSubmitting: PropTypes.bool,
+  values: PropTypes.shape({
+    title: PropTypes.string,
+    body: PropTypes.string,
+    field_experience_path: PropTypes.string
+  }).isRequired,
+  errors: PropTypes.shape({
+    title: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
+    field_experience_path: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.bool
+    ]),
+    body: PropTypes.oneOfType([PropTypes.string, PropTypes.bool])
+  }).isRequired,
+  touched: PropTypes.shape({
+    title: PropTypes.bool,
+    field_experience_path: PropTypes.bool,
+    body: PropTypes.bool
+  }).isRequired,
+  classes: PropTypes.shape({
+    textField: PropTypes.string.isRequired,
+    button: PropTypes.string.isRequired
+  }).isRequired,
+  user: PropTypes.shape({
+    authentication: PropTypes.shape({
+      accessToken: PropTypes.string.isRequired,
+      csrfToken: PropTypes.string.isRequired
+    }).isRequired
+  }).isRequired,
+  experiences: PropTypes.shape({
+    error: PropTypes.string,
+    items: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.string.isRequired,
+        title: PropTypes.string.isRequired,
+        body: PropTypes.PropTypes.shape({
+          value: PropTypes.string
+        }),
+        field_experience_path: PropTypes.string.isRequired
+      })
+    )
+  })
+};
 
-  /**
-   * Fetches the experience that belongs to the given slug.
-   */
-  getExperience = () => {
-    const {
-      experienceSlug,
-      experiences: { items }
-    } = this.props;
-    return experienceSlug
-      ? items.find(item => item.field_experience_path === experienceSlug)
-      : false;
-  };
+ExperienceForm.defaultProps = {
+  isSubmitting: false,
+  experiences: {
+    error: null,
+    items: []
+  }
+};
 
-  /**
-   * Handles dispatching create or update actions.
-   *
-   * @param {object} event - Submit event object.
-   */
-  handleSubmit = event => {
-    event.preventDefault();
+const FormikExperienceForm = withFormik({
+  displayName: 'ExperienceForm',
+  enableReinitialize: true,
+  mapPropsToValues: props => {
+    const { experienceSlug, experiences } = props;
+    const experience = parseExperience(experiences, experienceSlug);
+
+    const values = {
+      title: '',
+      field_experience_path: '',
+      body: ''
+    };
+
+    if (experience) {
```

Comment:
I default form values to empty strings for creation forms, then override those values if the current form is editorial (if an experience slug is provided)